### DISCUSSION TITLE
How do I `make standalone_package` into a separate directory?

### DISCUSSION THREAD

**Discussion by jimis** [Ericsson/codechecker#3560] [Q&A] [answered]
By default `make standalone_package` installs everything into `build/CodeChecker`.

Lets say I apply a patch and re-run `make standalone_package`. Is there some flag/variable to pass, so that the 2nd build goes to a completely separate destination, not affecting the first one?

**Comment by csordasmarton** [comment, marked as answer]
You can set the [`BUILD_DIR`](https://github.com/Ericsson/codechecker/blob/master/Makefile#L4) environment variable to a different one before calling this target and I think what will solve your problem:
```sh
BUILD_DIR=/home/username/codechecker/build2 make standalone_package
```

**Reply by jimis** [reply]
Thanks, it seems to work! 

On another note, the documentation could use some more info about installing to custom locations. It has an "install guide" section but it basically is a "build guide". Usually projects have a `make install` target that installs to a user-defined `prefix`. BUILD_DIR as used here is unusual, so it would be nice to be mentioned and even tested.

**Reply by jimis** [reply]
After a second check, some files seem to be missing from build2 directory:
```
Only in build/CodeChecker/bin: ldlogger
Only in build/CodeChecker/bin: merge-clang-extdef-mappings
Only in build/CodeChecker/bin: post-process-stats
Only in build/CodeChecker/bin: report-converter
Only in build/CodeChecker/bin: tu_collector
```
Executing CodeChecker fails with
```
ModuleNotFoundError: No module named 'codechecker_report_converter'
```

**Reply by jimis** [reply]
Found it! It seems BUILD_DIR must contain an absolute path, otherwise files are silently missing from the destination. I'll open an issue. EDIT: #3566
